scripts/zones/Windurst_Waters_[S]/npcs/Hampu-Kampu.lua

```diff
@@ -0,0 +1,54 @@
+-----------------------------------
+-- Area: Windurst Waters (S)
+--  NPC: Hampu-Kampu
+-- Type: Quest NPC
+-----------------------------------
+local ID = require("scripts/zones/Windurst_Waters_[S]/IDs")
+require("scripts/globals/quests");
+require("scripts/globals/keyitems");
+
+
+function onTrade(player,npc,trade)
+end
+
+function onTrigger(player,npc)
+    local sayItWithAHandbag = player:getQuestStatus(CRYSTAL_WAR, dsp.quest.id.crystalWar.SAY_IT_WITH_A_HANDBAG)
+    local sayItWithAHandbagCS = player:getVar("sayItWithAHandbagCS")
+
+    if (sayItWithAHandbag == QUEST_COMPLETED) then
+        player:startEvent(175)
+    elseif (player:hasKeyItem(dsp.ki.REPAIRED_HANDBAG) and sayItWithAHandbagCS == 4) then
```

Comment:
Again, since you're using a stage variable, you don't need to check the key item~

I think you can cut out all your KI checks. (Make sure you still give/remove the KI from the player though, even if you're not using them. Just don't need to check for them.)